### DISCUSSION TITLE
Integrate `redux` and` redux-persist`

### DISCUSSION THREAD

**Discussion by kimmy-wang** [vercel/next.js#13026] [Help]
When `next-i18next` is integrated in the application, clicking the language switch button will not report any errors; when integrating` redux` and `redux-persist`, clicking the language switch button will report an error. What causes this?

<img width="1324" alt="截屏2020-05-18 09 31 25" src="https://user-images.githubusercontent.com/19725091/82166823-45dc7100-98ec-11ea-9352-eaa6a9d0bd57.png">

[Websites](https://create-mock-data.now.sh/)

[repo](https://github.com/upcwangying/create-mock-data)

**Comment by rafinskipg** [comment]
Hey @upcwangying  I didn't try your code but i see you are executing `getInitialProps`. 

```

      let appProps = {}
      if (typeof App.getInitialProps === 'function') {
        appProps = await App.getInitialProps(appContext)
      }

````

I am pretty sure that this is automatically executed by Next, and you just need to receive them

Take a look at . https://nextjs.org/docs/api-reference/data-fetching/getInitialProps

**Reply by kimmy-wang** [reply]
In file `lib/with-redux-store.js`, there is already relevant code

<img width="890" alt="截屏2020-05-18 13 08 47" src="https://user-images.githubusercontent.com/19725091/82176240-c741fc80-9908-11ea-85d1-c1a8155103ac.png">

**Reply by rafinskipg** [reply]
Can you try to change App.getInitialProps(context) by this?

```javascript

 static async getInitialProps({ Component, ctx }) {
      // Get or Create the store with `undefined` as initialState
      // This allows you to set a custom default initialState
      const reduxStore = getOrCreateStore(initialState)

      // Provide the store to getInitialProps of pages
      ctx.reduxStore = reduxStore

      let appProps = {}
      if (typeof Component.getInitialProps === 'function') {
        appProps = await Component.getInitialProps(ctx)
      }

      return {
        ...appProps,
        initialReduxState: reduxStore.getState(),
      }
    }
```

**Reply by kimmy-wang** [reply]
<img width="793" alt="截屏2020-05-18 16 31 12" src="https://user-images.githubusercontent.com/19725091/82191509-34fc2180-9925-11ea-900e-ff48046f79aa.png">

<img width="997" alt="截屏2020-05-18 16 30 54" src="https://user-images.githubusercontent.com/19725091/82191528-3a596c00-9925-11ea-96dd-d5107bef5dc9.png">

**Reply by rafinskipg** [reply]
@upcwangying hey man.

Look, there is a package called next-redux-wrapper, they have implemented everything, you can take a look into the source code and find the solution:

https://github.com/kirill-konshin/next-redux-wrapper/blob/master/packages/wrapper/src/index.tsx


And here you have an example use case https://github.com/willianantunes/nextjs-playground/blob/master/src/pages/_app.js
